[npm_package] fails if there is no version field in package.json

https://travis-ci.com/github/adobe/asset-compute-integration-tests/jobs/412213493

```
comparator npm_package failed: { Error: Command failed: npm publish --dry-run
454npm ERR! package.json requires a valid "version" field
455
456npm ERR! A complete log of this run can be found in:
457npm ERR!     /home/travis/.npm/_logs/2020-10-28T00_41_48_948Z-debug.log
458
459    at ChildProcess.exithandler (child_process.js:294:12)
460    at ChildProcess.emit (events.js:198:13)
461    at maybeClose (internal/child_process.js:982:16)
462    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
463  killed: false,
464  code: 1,
465  signal: null,
466  cmd: 'npm publish --dry-run',
467  stdout: '',
468  stderr:
469   'npm ERR! package.json requires a valid "version" field\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /home/travis/.npm/_logs/2020-10-28T00_41_48_948Z-debug.log\n' }
```

--- a/lib/comparators/npm_package.js
+++ b/lib/comparators/npm_package.js
@@ -47,7 +47,17 @@ async function getSize(dir) {
 module.exports = {
 
     shouldRun: async function(beforeDir, afterDir) {
-        return fs.existsSync(path.join(afterDir, "package.json"));
+        const pkgJsonPath = path.join(afterDir, "package.json");
+        if (!fs.existsSync(pkgJsonPath)) {
+            return false;
+        }
+
+        const pkgJson = require(pkgJsonPath);
+        if (pkgJson.private || !pkgJson.version) {
+            debug("skpping npm_package: private package or no version in package.json");
+            return false;
+        }
+        return true;
     },
 
     compare: async function(before, after) {

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@adobe/sizewatcher",
-  "version": "1.0.0",
+  "version": "1.1.0",
   "description": "Warns if your pull requests introduce large size increases.",
   "main": "index.js",
   "bin": {

